### DISCUSSION TITLE
rangeTracks - isRemoveUnknown для жанров

### DISCUSSION THREAD

**Discussion by xvlad113x** [Chimildic/goofy#169] [Идея] [answered]
Снова я, здравствуйте!

Когда проверял жанры из своего плейлиста и добавлял их в rangeTracks заметил что у некоторых непопулярных исполнителей вовсе отсутствуют жанры. В документации же написано что rangeTracks отсеивает всех без жанровых исполнителей - [Описание параметров объектов](https://chimildic.github.io/goofy/#/guide?id=%d0%9e%d0%bf%d0%b8%d1%81%d0%b0%d0%bd%d0%b8%d0%b5-%d0%bf%d0%b0%d1%80%d0%b0%d0%bc%d0%b5%d1%82%d1%80%d0%be%d0%b2-%d0%be%d0%b1%d1%8a%d0%b5%d0%ba%d1%82%d0%be%d0%b2)

Но есть Musixmatch с [detectLanguage](https://chimildic.github.io/goofy/#/reference/filter?id=detectlanguage) в котором есть `isRemoveUnknown` при отсутствии текста песен позволяющий оставить треки.

Вопрос/просьба - добавите ли вы аналогичную опцию и для rangeTracks, для жанров?

**Comment by Chimildic** [comment, marked as answer]
Попробуйте следующий пример после обновления. Добавил ключ `isRemoveUnknownGenre`.
```js
Filter.rangeTracks(tracks, {
  artist: {
    genres: ['rock'],
    isRemoveUnknownGenre: false,
  },
});
```

**Reply by xvlad113x** [reply]
Начала появляться ошибка
![Безымянный](https://user-images.githubusercontent.com/44963372/151224655-6993c55d-6ce5-49b5-a3ec-51afd2a3cfc2.png)

**Reply by Chimildic** [reply]
Привередливый у вас список треков конечно. Теперь?

**Reply by xvlad113x** [reply]
> Привередливый у вас список треков конечно

![Безымянный](https://user-images.githubusercontent.com/44963372/151227780-4fc4c1e2-e482-4dbe-9a3c-6ccfe7031e21.png)
Не то слово 😂

> Теперь?

Все отлично, работает! Еще раз спасибо большое 🥳
